### PR TITLE
chore(TDI-44145): update commons-codec:1.14

### PR DESCRIPTION
**Pull request by ozhelezniak-talend**
* update commons-codec to 1.14 version

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44145

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [x] Other... Please describe: CVE

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/482/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/482/)

**Comment by Voievidko** [comment]
Two modules still use commons-codec:commons-codec:jar:1.10.
Please, check dep tree:

```
org.talend.components:components-api-service-rest:jar:0.29.0-SNAPSHOT
+- org.talend.daikon:daikon:jar:tests:0.31.11:test
|  +- org.apache.avro:avro:jar:1.8.1:compile
|  |  +- com.thoughtworks.paranamer:paranamer:jar:2.7:compile
|  |  +- org.xerial.snappy:snappy-java:jar:1.1.1.3:compile
|  |  +- org.apache.commons:commons-compress:jar:1.19:compile
|  |  \- org.tukaani:xz:jar:1.5:compile
|  +- org.talend.daikon:crypto-utils:jar:0.31.11:compile
|  |  +- commons-beanutils:commons-beanutils:jar:1.9.4:compile
|  |  |  \- commons-collections:commons-collections:jar:3.2.2:compile
|  |  \- org.apache.commons:commons-configuration2:jar:2.0:compile
|  +- org.apache.commons:commons-lang3:jar:3.8.1:compile
|  +- commons-codec:commons-codec:jar:1.10:compile


org.talend.components:components-api-service-rest-all-components:jar:0.29.0-SNAPSHOT
+- org.talend.daikon:daikon:jar:tests:0.31.11:test
|  +- org.apache.avro:avro:jar:1.8.1:compile
|  |  +- com.thoughtworks.paranamer:paranamer:jar:2.7:compile
|  |  +- org.xerial.snappy:snappy-java:jar:1.1.1.3:compile
|  |  +- org.apache.commons:commons-compress:jar:1.19:compile
|  |  \- org.tukaani:xz:jar:1.5:compile
|  +- org.apache.commons:commons-lang3:jar:3.8.1:compile
|  +- commons-codec:commons-codec:jar:1.10:compile
I suppose it comes from https://github.com/Talend/components/blob/master/services/components-api-service-rest-all-components/pom.xml#L6-L11.
```

This version of spring-boot-starter-parent use commons-codec:commons-codec:jar:1.10.
https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-parent/1.5.22.RELEASE

**Comment by ozhelezniak-talend** [comment]
Resolved the last two commons-codec:1.10 to commons-codec:1.14

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/483/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/483/)

File: services/components-api-service-rest-all-components/pom.xml [hunk 1]
```diff
@@ -42,6 +42,7 @@
         <docker-maven-plugin.version>0.26.0</docker-maven-plugin.version>
         <!-- Override beanutils version from spring-boot-starter-parent 1.5.1.RELEASE-->
         <commons-beanutils.version>1.9.4</commons-beanutils.version>
+        <commons-codec.version>1.14</commons-codec.version> <!-- spring boot sucks with its bom -->
```
Comment (Voievidko) [inline services/components-api-service-rest-all-components/pom.xml:45]: Please, rephrase comment.

File: services/components-api-service-rest/pom.xml [hunk 2]
```diff
@@ -34,6 +34,7 @@
         <groovy.version>2.5.6</groovy.version> <!-- spring boot sucks with its bom -->
         <osgi.core.version>6.0.0</osgi.core.version>
         <commons-io.version>2.5</commons-io.version>
+        <commons-codec.version>1.14</commons-codec.version> <!-- spring boot sucks with its bom -->
```
Comment (Voievidko) [inline services/components-api-service-rest/pom.xml:37]: Please, rephrase comment.

**Review by Voievidko — APPROVED** [review]
It is ok with few comments

File: services/components-api-service-rest-all-components/pom.xml (hunk 1 above)
Comment (ozhelezniak-talend) [reply services/components-api-service-rest-all-components/pom.xml:45]: sure, it was a copy of the previous same comment but I find another version with the same info

File: services/components-api-service-rest/pom.xml (hunk 2 above)
Comment (ozhelezniak-talend) [reply services/components-api-service-rest/pom.xml:37]: sure, it was a copy of the previous same comment but I find another version with the same info

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/485/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/485/)

**Pull request merged**